docs/corfu-queue-design.md

```diff
@@ -0,0 +1,15 @@
+The class `CorfuQueue` implements a persisted queue over the abstraction of a CorfuTable. Currently CorfuTable uses a HashMap<> to represent the materialized view of a distributed map. A HashMap does not carry a notion of ordering natively. However when implemented over the abstraction of a distributed shared log, the elements added to the map do, in fact, have an ordering imposed by their append operations into the global log. The class `CorfuQueue` attempts to expose this inherent ordering as an immutable logical FIFO Queue interface with three simple apis:
```

Comment:
This doesn't really use the `CorfuTable` and the FIFO order is dictated by the global commit order.

Reply:
Good point. Updating this doc now.